include/clients/rt/SineFeatureClient.hpp

```diff
@@ -0,0 +1,167 @@
+/*
+Part of the Fluid Corpus Manipulation Project (http://www.flucoma.org/)
+Copyright 2017-2019 University of Huddersfield.
+Licensed under the BSD-3 License.
+See license.md file in the project root for full license information.
+This project has received funding from the European Research Council (ERC)
+under the European Union’s Horizon 2020 research and innovation programme
+(grant agreement No 725899).
+*/
+#pragma once
+
+#include "../common/AudioClient.hpp"
+#include "../common/BufferedProcess.hpp"
+#include "../common/FluidBaseClient.hpp"
+#include "../common/FluidNRTClientWrapper.hpp"
+#include "../common/ParameterConstraints.hpp"
+#include "../common/ParameterSet.hpp"
+#include "../common/ParameterTrackChanges.hpp"
+#include "../common/ParameterTypes.hpp"
+#include "../../algorithms/public/SineFeatureExtraction.hpp"
+#include <tuple>
+
+namespace fluid {
+namespace client {
+namespace sinefeature {
+
+enum SineFeatureParamIndex {
+  kNPeaks,
+  kDetectionThreshold,
+  kSortBy,
+  kLogFreq,
+  kLogMag,
+  kFFT
+};
+
+constexpr auto SineFeatureParams = defineParameters(
+    LongParamRuntimeMax<Primary>("numPeaks", "Number of Sinusoidal Peaks", 10,
+                        Min(1),
+                        FrameSizeUpperLimit<kFFT>()),
+    FloatParam("detectionThreshold", "Peak Detection Threshold", -96, Min(-144),
```

Comment:
what were we smoking when we decided `detectionThreshold` was a decent name in `SinesClient`? Guess we're stuck with it now